Core/include/Acts/Seeding/GbtsTrackingFilter.hpp

```diff
@@ -33,75 +30,16 @@ struct GbtsEdgeState {
 
   GbtsEdgeState() = default;
 
-  explicit GbtsEdgeState(bool f) : m_initialized(f) {}
+  explicit GbtsEdgeState(bool f) : m_initialized(f) {};
 
-  void initialize(GbtsEdge<external_spacepoint_t>* pS) {
-    m_initialized = true;
+  ~GbtsEdgeState() = default;
 
-    m_J = 0.0;
-    m_vs.clear();
-
-    // n2->n1
-
-    float dx = pS->m_n1->m_spGbts.SP->x() - pS->m_n2->m_spGbts.SP->x();
-    float dy = pS->m_n1->m_spGbts.SP->y() - pS->m_n2->m_spGbts.SP->y();
-    float L = std::sqrt(dx * dx + dy * dy);
-
-    m_s = dy / L;
-    m_c = dx / L;
-
-    // transform for extrapolation and update
-    //  x' =  x*m_c + y*m_s
-    //  y' = -x*m_s + y*m_c
-
-    m_refY = pS->m_n2->m_spGbts.r();
-    m_refX =
-        pS->m_n2->m_spGbts.SP->x() * m_c + pS->m_n2->m_spGbts.SP->y() * m_s;
-
-    // X-state: y, dy/dx, d2y/dx2
-
-    m_X[0] =
-        -pS->m_n2->m_spGbts.SP->x() * m_s + pS->m_n2->m_spGbts.SP->y() * m_c;
-    m_X[1] = 0.0;
-    m_X[2] = 0.0;
-
-    // Y-state: z, dz/dr
-
-    m_Y[0] = pS->m_n2->m_spGbts.SP->z();
-    m_Y[1] = (pS->m_n1->m_spGbts.SP->z() - pS->m_n2->m_spGbts.SP->z()) /
-             (pS->m_n1->m_spGbts.r() - pS->m_n2->m_spGbts.r());
-
-    memset(&m_Cx[0][0], 0, sizeof(m_Cx));
-    memset(&m_Cy[0][0], 0, sizeof(m_Cy));
-
-    m_Cx[0][0] = 0.25;
-    m_Cx[1][1] = 0.001;
-    m_Cx[2][2] = 0.001;
-
-    m_Cy[0][0] = 1.5;
-    m_Cy[1][1] = 0.001;
-  }
-
-  void clone(const struct GbtsEdgeState& st) {
-    memcpy(&m_X[0], &st.m_X[0], sizeof(m_X));
-    memcpy(&m_Y[0], &st.m_Y[0], sizeof(m_Y));
-    memcpy(&m_Cx[0][0], &st.m_Cx[0][0], sizeof(m_Cx));
-    memcpy(&m_Cy[0][0], &st.m_Cy[0][0], sizeof(m_Cy));
-    m_refX = st.m_refX;
-    m_refY = st.m_refY;
-    m_c = st.m_c;
-    m_s = st.m_s;
-    m_J = st.m_J;
-    m_vs.clear();
-    m_vs.reserve(st.m_vs.size());
-    std::copy(st.m_vs.begin(), st.m_vs.end(), std::back_inserter(m_vs));
-
-    m_initialized = true;
-  }
+  void initialize(GbtsEdge* pS);
+  void clone(const struct GbtsEdgeState& st);
 
   float m_J{};
 
-  std::vector<GbtsEdge<external_spacepoint_t>*> m_vs;
+  std::vector<GbtsEdge*> m_vs;
```

Comment:
- do we have lifetime concerns with the raw pointer?
- should the `GbtsEdge` rather be `const GbtsEdge`?